spec/services/activities/import_from_csv_spec.rb

```diff
@@ -0,0 +1,88 @@
+require "rails_helper"
+
+RSpec.describe Activities::ImportFromCsv do
+  let(:organisation) { create(:organisation) }
+  let(:existing_activity) { create(:activity) }
+  let(:activity_attributes) do
+    {
+      "RODA ID" => existing_activity.roda_identifier_compound,
+      "Title" => "Here is a title",
+      "Description" => "Some description goes here...",
+      "Recipient Region" => "789",
+    }
+  end
+
+  subject { described_class.new(organisation: organisation) }
+
+  it "has an error if an Activity does not exist" do
+    activity_attributes["RODA ID"] = "FAKE RODA ID"
+
+    subject.import([activity_attributes])
+
+    expect(subject.errors.count).to eq(1)
+
+    expect(subject.errors.first.csv_row).to eq(2)
+    expect(subject.errors.first.column).to eq(:roda_id)
+    expect(subject.errors.first.value).to eq("FAKE RODA ID")
+    expect(subject.errors.first.message).to eq(I18n.t("importer.errors.activity.not_found"))
+  end
+
+  it "updates an existing activity" do
+    subject.import([activity_attributes])
+
+    expect(subject.errors.count).to eq(0)
+
+    expect(existing_activity.reload.title).to eq(activity_attributes["Title"])
+    expect(existing_activity.description).to eq(activity_attributes["Description"])
+    expect(existing_activity.recipient_region).to eq(activity_attributes["Recipient Region"])
+  end
+
+  it "ignores any blank columns" do
```

Comment:
Hopefully we'll never see a case where a column *should* be empty and that should be updated on the Activity(!)